prototype2/gdgem/nmx/Readout.h

```diff
@@ -1,40 +1,98 @@
 /* Copyright (C) 2017-2018 European Spallation Source, ERIC. See LICENSE file */
 //===----------------------------------------------------------------------===//
 ///
+///                              WARNING
+///
+///                              ACHTUNG
+///
+///                              AVISO
+///
+///                              ADVARSEL
+///
+///                              DĖMESIO
+///
+///                              UWAGA
+///
+///
+///
+///          MODIFY THIS FILE ONLY AFTER UNDERSTANDING THE FOLLOWING
+///
+///
+///   Any changes to non-static variable definitions will likely break h5 file
+/// writing and compatibility. If you rename, reorder or change the type of any
+/// of the member variables, you MUST:
+///    A) Increment FormatVersion by 1
+///    B) Ensure the hdf5 TypeTrait maps the struct correctly
+///
+/// If you cannot ensure the above, consult someone who can.
+///
 /// \file
 ///
+/// \brief Readout struct for NMX event formation
+///
 //===----------------------------------------------------------------------===//
 
 #pragma once
 
-#include <cinttypes>
-
-// TODO: initialize values to 0?
-struct Readout
-{
-  uint8_t fec;
-  uint8_t chip_id;
-  uint32_t bonus_timestamp;
-  uint32_t srs_timestamp;
-  uint16_t channel;
-  uint16_t bcid;
-  uint16_t tdc;
-  uint16_t adc;
-  float ChipTimeNs;
-  bool over_threshold;
-
-  bool operator==(const Readout& other) const
-  {
+#include <common/DumpFile.h>
+
+struct __attribute__ ((packed)) Readout {
+  // \todo use constexpr string_view when c++17 arrives
+  static std::string DatasetName() { return "srs_hits"; }
+  static uint16_t FormatVersion() { return 0; }
+
+  // !!! DO NOT MODIFY BELOW - READ HEADER FIRST !!!
+  uint8_t fec{0};
+  uint8_t chip_id{0};
+  uint32_t bonus_timestamp{0};
+  uint32_t srs_timestamp{0};
+  uint16_t channel{0};
+  uint16_t bcid{0};
+  uint16_t tdc{0};
+  uint16_t adc{0};
+  bool over_threshold{false};
+  float ChipTimeNs{0.0};
```

Comment:
Why does this field get CamelCase?

Reply:
This field appears to have been added at a different time by a different author. I did not want to mess with it. Should we LLVMify all of this to make it all CamelCase?

Reply:
yeah, it was probably me remembering for once the llvm code standard. I am actually not infavor of everything being camel case. Feel free to tolower() it for consistency.

Reply:
Let's leave this inconsistency for now. We can revamp this struct and have a new version of the dump format when a) our understanding of the SRS data improves, and b) we have some reference data from vmm3 which will replace the old files.